resources/assets/js/videos/models/Annotation.vue

```diff
@@ -283,12 +283,14 @@ export default Vue.extend({
                 this.points.splice(index, 1);
 
                 // Remove null (gap filler) as first/last element to prevent validation errors
-                if (this.frames[0] === null) {
+                // Multiple consecutive 'null' can occur after removing elements of linked anntations
+                // in specific order
+                while (this.frames[0] === null) {
                     this.frames.shift();
                     this.points.shift();
                 }
 
-                if (this.frames.at(-1) === null) {
+                while (this.frames.at(-1) === null) {
                     this.frames.pop();
                     this.points.pop();
                 }
```

Comment:
Why is this also needed? As I understand it, the new bug you found is solved by updating `validatePoints()` but why this, too? Maybe you can elaborate in the comment (what "specific order"?).

Reply:
This was another bug, where multiple consecutive nulls can occur, if you delete 
1. first/last annotation of an multiframe annotation
2. first/last annotation of an single frame annotation
After removing the first, a null is added in place of the removed element, which leads to two nulls at the array start/end.

Reply:
Sorry but I still don't get it. Could you please tell me exactly what I have to do to reproduce this?

Reply:
Sorry, this is my mistake. I mixed up the order. You can see in the video, if you first delete the multi frame annotation frame by frame and then a linked single frame annotation, an error is thrown due to the fact, that there are two nulls at the start. The if() statement removes null only once, which is why the second null remains in the array.

[Bildschirmaufzeichnung vom 12.01.2024, 09:48:40.webm](https://github.com/biigle/core/assets/72259845/d01adcdb-7494-4252-9648-d21e9716963b)

Reply:
Unfortunately no. You can go on and add more single frame annotations, link and delete them such that more nulls are added.

Reply:
But when each frame is deleted, the logic above would also delete one `null`, right? So the `null`'s should not accumulate.

Reply:
Here you can see, how I ended up with three nulls. 

[Bildschirmaufzeichnung vom 15.01.2024, 07:40:32.webm](https://github.com/biigle/core/assets/72259845/69ab55a2-676a-4491-b654-402ff9a86a1a)

A null is only deleted if it is the first or last element. If you accumulate them in the middle, they are not deleted, which is why I used a while loop. If you don't like while-loops, I could identify start and end index of all nulls and delete those areas.

Reply:
I'm still convinced by the logic above :smile: Each time you delete a keyframe, a `null` would be deleted, too.

Your (current) while loop only trims the `null`s at the start and end, not in the middle (as demonstrated in your video). With the logic above I think it would work in all cases and no loop is needed any more.

Reply:
In my videos I use the code with the if-statements. Maybe I should have said this before :sweat_smile:

Reply:
Interesting. Can you please push it? Then I'll have a look myself.